compiler/rustc_traits/src/normalize_erasing_regions.rs

```diff
@@ -31,8 +31,14 @@ fn normalize_generic_arg_after_erasing_regions<'tcx>(
                     None,
                 );
 
-                let normalized_value = infcx.resolve_vars_if_possible(normalized_value);
-                infcx.tcx.erase_regions(normalized_value)
+                let resolved_value = infcx.resolve_vars_if_possible(normalized_value);
+                // It's unclear when `resolve_vars` would have an effect in a
+                // fresh `InferCtxt`. If this assert does trigger, it will give
+                // us a test case.
+                debug_assert_eq!(normalized_value, resolved_value);
```

Comment:
I'm ok with this, but I'm not sure if this is what @lcnr meant -- I suspect they meant to move the `resolve_vars_if_possible` into the debug assert as well. I guess that I mildly prefer this in that I don't think it's generally meant to be an invariant that functions will return "fully resolved" types that don't contain inference variables.

Reply:
I think the approach by @jyn514 is good as it doesn't even change the behavior, so it most definity can't break anything

I would keep it as is, even if it isn't what i originally intended